obelisk/cpp/obelisk_cpp/include/obelisk_node.h

```diff
@@ -0,0 +1,451 @@
+                "This may cause certain API incompatibilities.");
+        }
+
+        return rclcpp_lifecycle::LifecycleNode::create_subscription<
+            MessageT, CallbackT, AllocatorT, SubscriptionT,
+            MessageMemoryStrategyT>(topic_name, qos, std::move(callback),
+                                    options, msg_mem_strat);
+    };
+
+  protected:
+    /**
+     * @brief Create a subscriber from a configuration string.
+     *
+     * @param config the configuration string
+     * @param callback the callback function (generall of the form
+     * std::bind(&MyFunc, this, _1))
+     * @return the subscription
+     */
+    template <
+        typename MessageT, typename CallbackT,
+        typename AllocatorT    = std::allocator<void>,
+        typename SubscriptionT = rclcpp::Subscription<MessageT, AllocatorT>>
+    std::shared_ptr<SubscriptionT>
+    CreateSubscriptionFromConfigStr(const std::string& config,
+                                    CallbackT&& callback) {
+        // Parse the configuration string
+        const auto config_map                = ParseConfigStr(config);
+        const std::string topic              = GetTopic(config_map);
+        const int depth                      = GetHistoryDepth(config_map);
+        const bool non_obelisk               = !GetIsObeliskMsg(config_map);
+
+        rclcpp::CallbackGroup::SharedPtr cbg = nullptr; // default group
+        try {
+            // Get the callback group based on the string name
+            cbg = callback_groups_.at(GetCallbackGroupName(config_map));
+        } catch (const std::exception& e) {
+        }
+
+        rclcpp::SubscriptionOptions options;
+        options.callback_group = cbg;
+
+        // Create the subscriber
+        return create_subscription<MessageT>(topic, depth, std::move(callback),
+                                             non_obelisk, options);
+    }
+
+    /**
+     * @brief Create a publisher from a configuration string
+     *
+     * @param config the configuration string
+     * @return the publisher
+     */
+    template <typename MessageT, typename AllocatorT = std::allocator<void>>
+    std::shared_ptr<rclcpp_lifecycle::LifecyclePublisher<MessageT, AllocatorT>>
+    CreatePublisherFromConfigStr(const std::string& config) {
+        // Parse the configuration string
+        const auto config_map   = ParseConfigStr(config);
+        const std::string topic = GetTopic(config_map);
+        const int depth         = GetHistoryDepth(config_map);
+        const bool non_obelisk  = !GetIsObeliskMsg(config_map);
+
+        return create_publisher<MessageT>(topic, depth, non_obelisk);
+    }
+
+    /**
+     * @brief Create a wall timer from a configuration string
+     *
+     * @param config the configuration string
+     * @param callback the callback function
+     *
+     * @return the timer
+     */
+    template <typename DurationT = std::milli, typename CallbackT>
```

Comment:
does this mean you take milliseconds by default? I thought you were doing something finer (i think we should do at least micro)

Reply:
yea so this is actually the ros default which is why I ultimately went with it, but I can change it to micro seconds.